File: Polimorfismo/Ej1/Videojuego.java
package Polimorfismo.Ej1;

public class Videojuego {
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;

    // Constructor por defecto
    public Videojuego() {
        this.nombre = "Desconocido";
        this.plataforma = "Indefinida";
        this.cantidadJugadores = 1;
    }

    // Constructor con nombre y plataforma
    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 1;
    }

    // Constructor completo
    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    public void mostrar() {
        System.out.println("Videojuego: " + nombre);
        System.out.println("Plataforma: " + plataforma);
        System.out.println("Jugadores: " + cantidadJugadores);
    }

    // Sobrecarga de método: agregar un solo jugador
    public void agregarJugadores() {
        this.cantidadJugadores++;
    }

    // Sobrecarga de método: agregar varios jugadores
    public void agregarJugadores(int cantidad) {
        this.cantidadJugadores += cantidad;
    }
}

File: Polimorfismo/Ej1/Videojuego.py
class Videojuego:
    def __init__(self, nombre="Desconocido", plataforma="Indefinida", cantidad_jugadores=1):
        self.nombre = nombre
        self.plataforma = plataforma
        self.cantidad_jugadores = cantidad_jugadores

    def mostrar(self):
        print(f"Videojuego: {self.nombre}")
        print(f"Plataforma: {self.plataforma}")
        print(f"Jugadores: {self.cantidad_jugadores}")

    def agregar_jugadores(self, cantidad=1):
        self.cantidad_jugadores += cantidad


# Instancias y prueba directa
juego1 = Videojuego("FIFA 23", "PC", 4)
juego2 = Videojuego("Zelda", "Nintendo Switch", 1)

juego1.mostrar()
print()
juego2.mostrar()
print()

# Agregar jugadores (simulando sobrecarga)
juego1.agregar_jugadores(2)
juego2.agregar_jugadores(3)

print("\nDespués de agregar jugadores:")
juego1.mostrar()
print()
juego2.mostrar()
print()
